docs/flows.md

```diff
@@ -0,0 +1,90 @@
+5. Your own custom auth middleware validates the authorization
+  - if valid then proceed to the service before hooks or any other auth middleware.
+  - if invalid set `data.authenticated = false` and move on to next middleware
+6. Service `before` hook, `isAuthenticated` checks to see if the request is authenticated.
+    - If it is proceed to next hook or service method
+    - If it isn't it rejects with a `NotAuthenticated` error
+7. Service `before` hook, `isPermitted` checks to see if the request is authenticated.
+    - If it is proceed to next hook or service method
+    - If it isn't it rejects with a `Forbidden` error
+8. Service method is called and returns through after hooks
+9. Response is sent back to the client
+
+## Logout (Not Expired Token)
+
+1. Client `DELETE` to `/authentication/:authorization_id` service with a valid JWT access token.
+2. Server verifies the token
+3. Looks up the authorization
+4. Calls `remove` on the `authorization` service
+5. Calls `app.emit('logout)`
+6. Responds with success
+7. Client can do whatever it wants (typically redirect to somewhere)
+
+## Logout (Expired Token)
+
+1. Client `DELETE` to `/authentication/:authorization_id` service with a expired JWT access token.
+2. Server recognizes the token is expired (removes or flags authorization as revoked/expired)
+3. Returns a `TokenExpired` error to the client
+4. Client can do whatever it wants (typically redirect to somewhere)
+
+## Authorization Renewal/Extension without Refresh Token
+An expired access token (JWT) is passed via an `Authorization` header **or** is attached to the connected socket. These are requests made to anything but the `/authentication` service.
+
+1. Client sends expired JWT access token or it's attached to the connected socket
+2. Server verifies JWT and determines it is expired
+  - returns a `NotAuthenticated` error with a "token expired" message to the client (current)
+  - **Proposed Change:** Return a special error type of `TokenExpired` and flags authorization as expired/revoked or delete it altogether.
+3. Client handles the error however they want (typically a redirect to login again)
+  - The client could also send a refresh token, or ClientId and ClientSecret to automatically get a new JWT access token depending on what is applicable to your app.
+
+## Authorization Renewal/Extension via Refresh Token
+An expired access token (JWT) is passed via an `Authorization` header **or** is attached to the connected socket. These are requests made to anything but the `/authentication` service.
+
+1. Client sends expired JWT access token or it's attached to the connected socket
+2. Server verifies JWT and determines it is expired
+  - returns a `NotAuthenticated` error with a "token expired" message to the client (current)
+  - **Proposed Change:** Return a special error type of `TokenExpired` and flags authorization as expired/revoked
```

Comment:
Actually I would think 401 Not Authenticated would be the correct response (at least HTTP status code wise). The refresh token is effectively a form of credentials (so login, but automated)